### PR TITLE
Add some extra tests for untested areas of the codebase.

### PR DESCRIPTION
**Pull request by dburgener**

### CODE REVIEW

File: lalrpop/src/lr1/report/test.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    let grammar = normalized_grammar(GRAMMAR_TEXT);
+    let _tls = Tls::install(
+        Rc::new(Session::new()),
+        Rc::new(FileText::new("".into(), "".into())),
+    );
+    let _lr1_tls = Lr1Tls::install(grammar.terminals.clone());
+    let start_nt = grammar.start_nonterminals.keys().next().unwrap().clone();
+    generate_report(&mut output, &lr1::build_states(&grammar, start_nt)).unwrap();
+
+    let report = String::from_utf8(output).unwrap();
```
Comment (Pat-Lafon) [inline lalrpop/src/lr1/report/test.rs:34]: Nit: Could output just have been a string to begin with instead of a Vec<u8> that gets converted to a string?

**Review by Pat-Lafon — APPROVED** [review]
More tests are always good!

File: lalrpop/src/lr1/report/test.rs (hunk 1 above)
Comment (dburgener) [reply lalrpop/src/lr1/report/test.rs:34]: It could not, because Strings don't implement `io::Write`, and `generate_report()`s output buffer must do so.

Comment (Pat-Lafon) [reply lalrpop/src/lr1/report/test.rs:34]: Ah, because it needs to be valid UTF-8 for strings and io::Write lets you write bytes. Whoops

**Pull request merged**